src/main/kotlin/io/github/nomisrev/repo/UserPersistence.kt

```diff
@@ -48,7 +50,10 @@ interface UserPersistence {
     image: String?
   ): Either<DomainError, UserInfo>
 
-  suspend fun unfollowProfile(followedUsername: String, followerId: UserId): Unit
+  suspend fun unfollowProfile(followedUsername: String, followerId: UserId)
+
+  context(Raise<DomainError>)
+  suspend fun followProfile(followedUsername: String, followerId: UserId)
```

Comment:
I am keeping context receivers for the POC PR.

```suggestion
  suspend fun followProfile(followedUsername: String, followerId: UserId): Either<DomainError, Unit>
```